src/components/data-table/ha-data-table.ts

```diff
@@ -321,7 +322,7 @@ export class HaDataTable extends LitElement {
                         ></ha-svg-icon>
                       `
                     : ""}
-                  <span>${column.title}</span>
+                  <span aria-label=${column.ariaLabel}>${column.title}</span>
```

Comment:
Should this be on the `span` or on the parent `div.mdc-data-table__header-cell`?

Reply:
By re-reading code and documentations, the div looks like better as it it has the role and can be interactive.
I'm going to modify it.